core/src/main/java/org/apache/iceberg/MetadataTableUtils.java

```diff
@@ -100,9 +100,8 @@ public static Table createMetadataTableInstance(
       TableIdentifier baseTableIdentifier,
       TableIdentifier metadataTableIdentifier,
       MetadataTableType type) {
-    String baseTableName = BaseMetastoreCatalog.fullTableName(catalogName, baseTableIdentifier);
-    String metadataTableName =
-        BaseMetastoreCatalog.fullTableName(catalogName, metadataTableIdentifier);
+    String baseTableName = CatalogUtil.fullTableName(catalogName, baseTableIdentifier);
```

Comment:
this doesn't have to change because we need to keep `BaseMetastoreCatalog.fullTableName` so that we don't break the API